docs(SRV): add jobs documentation MTA-3846 (#2367)
scaleway/docs-content: serverless/jobs/reference-content/difference-with-functions-containers.mdx

content:
  h1: Differences between Serverless Jobs, Containers and Functions
  paragraph: This page explains the differences between Scaleway's Serverless jobs, Serverless Functions and Serverless Containers
tags: job container functions differences vs docker image
dates:
  validation: 2023-11-13
  posted: 2023-11-13
categories:
  - serverless
  - jobs
---

<Lightbox src="serverless-decision-tree.webp" alt="" />

## Differences between Functions, Containers and Jobs

Serverless Jobs are designed for batch computing workloads particularly large-scale and asynchronous tasks, like data processing, image rendering, or complex computations.

Choose Scaleway Serverless Jobs:

- for long running tasks and resource-intensive applications
- to handle a large number of jobs concurrently
- if you do not need to expose your service on a port
- for asychronous processing

<Message type="note">
  To run as a Serverless Job, your applications must be containerized, which offers you a lot of choice when configuring your runtime environments.
</Message>

Choose Scaleway Serverless Functions:

> - for for stateless and short-lived workloads (< 15 minutes)

Reviewer: ```suggestion
- for stateless and short-lived workloads (< 15 minutes)
```

---
meta:
  title: Differences between Serverless Jobs, Containers and Functions
  description: This page explains the differences between Scaleway's Serverless jobs, Serverless Functions and Serverless Containers
content:
  h1: Differences between Serverless Jobs, Containers and Functions
  paragraph: This page explains the differences between Scaleway's Serverless jobs, Serverless Functions and Serverless Containers
tags: job container functions differences vs docker image
dates:
  validation: 2023-11-13
  posted: 2023-11-13
categories:
  - serverless
  - jobs
---

<Lightbox src="serverless-decision-tree.webp" alt="" />

## Differences between Functions, Containers and Jobs

> Serverless Jobs are designed for batch computing workloads particularly large-scale and high-throughput jobs, like data processing, image rendering, or complex computations.

Reviewer 2: ```suggestion
Serverless Jobs are designed for batch computing workloads particularly large-scale asynchronous tasks, like data processing, image rendering, or complex computations.
```

Choose Scaleway Serverless Jobs:

> - for long running tasks and resource intensive applications

Reviewer 2: ```suggestion
- for long running tasks and resource-intensive applications
```

> - to handle a large number of jobs efficiently

Reviewer 2: ```suggestion
- to handle a large number of jobs concurrently
```

- if you do not need to expose your service on a port
- for asychronous processing

<Message type="note">

>   Serverless Jobs need your applications to be containerized, and therefore provide more flexibility in terms of runtime environments.

Reviewer 2: ```suggestion
  To run as a Serverless Job, your applications must be containerized, which offers you a lot of choice when configuring your runtime environments.
```

</Message>

Choose Scaleway Serverless Functions:

> - for stateless and small workloads

Reviewer 2: ```suggestion
- for stateless and short-lived workloads (<15 minutes)
```

- for event-driven tasks
- for optimum performance in lightweight scenarios

> - to scale tasks automatically

Reviewer 2: ```suggestion
- for auto-scaling applications
```

> - for short execution time (< 15 min)

Reviewer 2: ```suggestion
- for web-facing applications and web servers
```

<Message type="note">

>   Serverless Functions are limited to available runtimes.

Reviewer 2: ```suggestion
  Serverless Functions can only be written in languages available as Serverless Function runtimes <INSERT LINK TO FUNCTIONS PAGE>
```

</Message>

Choose Scaleway Serverless Containers:

> - for stateless or stateful containerized applications like  microservices, APIs, and more

Reviewer 2: ```suggestion
- for stateless, short-lived applications (<15 minutes)
- for web-facing applications such as web servers, APIs and microservices
```

- for event-driven tasks

> - if you need more control than with Serverless Functions

Reviewer 2: ```suggestion
- if your programming language is not supported as a Serverless Function runtime
- if you need more control over your runtime environment with Serverless Functions
```

> - to scale tasks automatically

Reviewer 2: ```suggestion
- for auto-scaling workloads
```

> - for short execution time (< 15 min)

Reviewer 2: ```suggestion

```

<Message type="note">

>   Serverless Containers need your applications to be containerized, and therefore provide more flexibility in terms of runtime environments.

Reviewer 2: ```suggestion
  To run as a Serverless Container, your application must be containerized, giving you a lot of flexibility to configure your runtime environments.
```
